src/ASM/AlgEqSystem.C

```diff
@@ -66,7 +66,10 @@ bool AlgEqSystem::init (LinAlg::MatrixType mtype, const LinSolParams* spar,
     }
 
     PROFILE("Pre-assembly");
-    A[i]._A->initAssembly(sam,preAssemblyFlag);
+    if (i == 0)
+      A[i]._A->initAssembly(sam,preAssemblyFlag);
+    else
+      A[i]._A = A[0]._A->copy();
```

Comment:
So this seems to work, but the _A matrix for i > 0 has then already been created unnecessarily, so maybe L59 should be

    if (i == 0 && !A[0]._A)

or put the create step before the loop to make it even more clear